### PR TITLE
Add logging to AgentApi.cs

### PR DESCRIPTION
**Pull request by mvu02**

### CODE REVIEW

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs [hunk 1]
```diff
@@ -4519,46 +4529,79 @@ public string BaseUrl
                     request_.Content = content_;
                     request_.Method = new System.Net.Http.HttpMethod("POST");
                     request_.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
-
```
Comment (tchitnis) [inline src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: same as above

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs [hunk 2]
```diff
@@ -4510,7 +4519,8 @@ public string BaseUrl
                     // https://tools.ietf.org/html/rfc7578#section-4.4
                     // The nswag generator does not include this header so this patch forces
                     // that in.
-                    if (attachment != null) {
+                    if (attachment != null)
```
[the remark this answers is not in the document]
Comment (mvu02) [reply src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: I didn't make the formatting changes, my VS did it automatically.

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs (hunk 1 above)
Comment (mvu02) [reply src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: It was auto formatted by my VS.

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs [hunk 3]
```diff
@@ -4519,46 +4529,79 @@ public string BaseUrl
...
                     var url_ = urlBuilder_.ToString();
                     request_.RequestUri = new System.Uri(url_, System.UriKind.RelativeOrAbsolute);
                     PrepareRequest(client_, request_, url_);
-    
+
+                    var requestDetails = new Dictionary<string, string>()
+                    {
+                        {"url", url_}
+                    };
+                    _log.LogDebug("Attempting to send the request to symphony.", requestDetails);
```
Comment (tchitnis) [inline src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: shouldn't this be LogStructuredDebug?

Comment (mvu02) [reply src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: We actually don't use CollabLibrary in RestApiClient, so we don't have access to that extension. We only use the built in logging from Microsoft.

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs [hunk 4]
```diff
@@ -4519,46 +4529,79 @@ public string BaseUrl
...
+                        _log.LogDebug("Successfully processed the response from symphony.");
+
+                        var status_ = ((int) response_.StatusCode).ToString();
+
+                        var statusDetails = new Dictionary<string, string>()
+                        {
+                            {"status", status_}
+                        };
+
+                        var responseDataDetails = new Dictionary<string, string>()
```
Comment (tchitnis) [inline src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: why cant we do this after we have the responseData ?

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs [hunk 5]
```diff
@@ -4519,46 +4529,79 @@ public string BaseUrl
...
+                            _log.LogDebug("Logging response data.", responseDataDetails);
+
+                            var result_ = default(V4Message);
                             try
                             {
                                 result_ = Newtonsoft.Json.JsonConvert.DeserializeObject<V4Message>(responseData_, _settings.Value);
-                                return result_; 
-                            } 
-                            catch (System.Exception exception_) 
+                                return result_;
```
Comment (tchitnis) [inline src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: Can we get rid of these unnecessary changes VS has added?

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs (hunk 4 above)
Comment (tchitnis) [reply src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: Ah I see why now.

Comment (mvu02) [reply src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: Then we would be creating a new object for every if case.

Comment (tchitnis) [reply src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: do we really need a dictionary here if the responseData is always a string?

File: src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs [hunk 6]
```diff
@@ -4625,6 +4667,11 @@ public string BaseUrl
                     }
                 }
             }
+            catch (Exception e)
+            {
+                _log.LogError(0, e, "An error occured while trying to send the request to symphony.");
```
Comment (tchitnis) [inline src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs]: I think you might have to go something like this here as well:

_log.LogError("Unable to launch monster data flusher, ex={0}", ex);

**Comment by tchitnis** [comment]
I think this looks good now.
